src/Bicep.Cli.IntegrationTests/Files/BuildParamsCommandTests/Extends/shared.json

```diff
@@ -0,0 +1,5 @@
+{
+  "parametersJson": "{\n  \"$schema\": \"https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#\",\n  \"contentVersion\": \"1.0.0.0\",\n  \"parameters\": {\n    \"stringParam\": {\n      \"value\": \"foo\"\n    }\n  }\n}",
+  "templateJson": null,
+  "templateSpecId": null
```

Comment:
Have you given any thought to how to manage the transition to this output format in PowerShell and the CLI? It looks like returning `null` for both `templateJson` and `templateSpecId` will cause both [the `Az` PS module](https://github.com/Azure/azure-powershell/blob/1e7af8e1c5414893cb8a2a243f2eabf28118680e/src/Resources/ResourceManager/Implementation/CmdletBase/DeploymentCmdletBase.cs#L415-L418) and [the CLI](https://github.com/Azure/azure-cli/blob/7bb63890632021456d3dacf55cf709e4aede045b/src/azure-cli/azure/cli/command_modules/resource/custom.py#L1155) to fail with somewhat cryptic messages about an invalid file path or invalid JSON.

What if the Bicep CLI instead returned a non-zero exit code in cases where a .bicepparam file with `using none` is compiled directly? That would integrate better with downstream tooling.

Reply:
I had a chat with @anthony-c-martin about this. we're going to publish this feature first, then we'll update the azure cli and powershell modules to support this.